paasta_tools/kubernetes_tools.py

```diff
@@ -445,16 +447,27 @@ def __init__(self) -> None:
             ),
             fset=_set_disrupted_pods,
         )
-
-        self.deployments = kube_client.AppsV1Api()
-        self.core = kube_client.CoreV1Api()
-        self.policy = kube_client.PolicyV1beta1Api()
-        self.apiextensions = kube_client.ApiextensionsV1beta1Api()
-        self.custom = kube_client.CustomObjectsApi()
-        self.autoscaling = kube_client.AutoscalingV2beta2Api()
-        self.rbac = kube_client.RbacAuthorizationV1Api()
+        if not component:
+            # If we don't get an explicit component set via constructor,
+            # try to find it by looking back in the stack, and getting `__file__` from
+            # the context calling this constructor
+            # Normally, `__module__` would make more sense, but since we have a lot of
+            # single scripts we directly call, that would be set to `__main__` most of the time.
+            current = currentframe()
+            parent = current.f_back
+            component = parent.f_globals["__file__"]
```

Comment:
is there any case where `__file__` can be missing from`globals`? probably more useful to do `.get('__file__', 'unknown')` instead?